core/shared/src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -40,12 +40,22 @@ trait Queue[F[_], A] { self =>
   /** Dequeue one `A` from this queue. Completes once one is ready. */
   def dequeue1: F[A]
 
+  /** Dequeue all available `A`s from this queue. Completes once any values are ready. */
+  def dequeueAvailable1: F[Vector[A]]
+
   /** Like `dequeue1` but provides a way to cancel the dequeue. */
   def cancellableDequeue1: F[(F[A], F[Unit])]
 
+  /** Like `dequeueAvailable1` but provides a way to cancel the dequeue. */
+  def cancellableDequeueAvailable1: F[(F[Vector[A]], F[Unit])]
+
   /** Repeatedly call `dequeue1` forever. */
   def dequeue: Stream[F, A] = Stream.bracket(cancellableDequeue1)(d => Stream.eval(d._1), d => d._2).repeat
 
+  /** Repeatedly call `dequeueAvailable1` forever. */
+  def dequeueAvailable: Stream[F, A] =
```

Comment:
Yeah, actually, nevermind. Wasn't really thinking too carefully. :)

I didn't mean get rid of `dequeue1`, I was thinking just have `dequeue` (which returns a `Stream`) use `dequeueAvailable`. I was thinking that since it is returning the entire stream of dequeued elements, there shouldn't need to be any pushback... you just `dequeueAvailable` to obtain a `Vector[A]`, then `Stream.emits` that. BUT then I remembered stuff like `q.dequeue.take(4)` -- that could consume more elements from the queue than are needed if it were implemented in terms of `dequeueAvailable`. So having both versions seems fine.

I did not do a super thorough review but looks 👍  to me. @pchlupacek / @mpilquist and/or @djspiewak may want to weigh in before merging.